docs/src/contribute/contribution_standards.md

```diff
@@ -1,159 +1,180 @@
 # Contribution Standards
 
-## Formatting and Style
+## Overview
 
-We install various style and formatting enforcers in our `pre-commit` command which is configured in `.pre-commit-config.yaml` in the root of the repository. These include `ruff` for python and `terraform` standards. 
+This document outlines the standards and practices for contributing to the MATRIX drug repurposing platform. These guidelines help maintain code quality, consistency, and facilitate collaboration across our development team.
+
+## Technology Stack & Language Standards
+
+### Languages to be used
+
+- **Python 3.11+** - Primary language for all business logic, data processing, and ML pipelines
+  - Package management via [uv](https://docs.astral.sh/uv/) (modern Python package manager)
+  - Formatting and linting with [Ruff](https://docs.astral.sh/ruff/)
+- **JavaScript/Node.js** - Dashboard applications using [Evidence.dev](https://evidence.dev/) framework only
+- **Shell/Bash** - Infrastructure automation and utility scripts
+- **SQL** - [BigQuery](https://cloud.google.com/bigquery/docs) analytics and data warehouse operations
+- **HCL** - [Terraform](https://www.terraform.io/docs)/[Terragrunt](https://terragrunt.gruntwork.io/docs/) infrastructure definitions
+
+### Restricted Languages
+
+No other programming languages should be introduced without explicit team approval and architectural justification. This includes but is not limited to Rust, Go (for application code), Java, or C++.
+
+## Development Workflow
+
+### Git Workflow
+
+- **Never commit directly** to `main` or `develop` branches
+- Always work on feature branches with descriptive names:
+  - `feat/descriptive-name` for new features
+  - `fix/descriptive-name` for bug fixes  
+  - `dev/your-name/task` for experimental work
+- Create [draft pull requests](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests) early for feedback and collaboration
+- All PRs require core maintainer approval before merging
+- CI checks must pass (linting, testing, security scans) before review
```

Comment:
One thing that will get increasingly important for a large monorepo is finding commits. I suggest you start introducing at least a barebones guide for commit messages, like https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/. You are already quite good at that as an org, but there is always room to improve this.

You might want to add "no PRs without issues" but thats just me.

Reply:
Also I know this goes without saying but:

- NEVER ever amend commits you have already pushed to your branch.

Reply:
IMO commits on branches can be `bump` etc but the main branch needs to be clean (and it mostly is, since github names the merge commit as the PR was named)